paprika/amber.py

```diff
@@ -470,30 +503,63 @@ def run(self, soft_minimize=False, overwrite=False, fail_ok=False):
             if self.mden is not None:
                 exec_list += ["-e", self.mden]
 
-            logger.debug("Exec line: "+" ".join(exec_list))
+            logger.debug("Exec line: " + " ".join(exec_list))
+
+            # Add CUDA_VISIBLE_DEVICES variable to env dict
+            if self.gpu_devices:
+                if "CUDA_VISIBLE_DEVICES" not in os.environ.keys():
+                    os.environ = dict(
+                        os.environ, CUDA_VISIBLE_DEVICES=str(self.gpu_devices)
+                    )
+                else:
+                    os.environ["CUDA_VISIBLE_DEVICES"] = self.gpu_devices
+
+            # Set the Plumed kernel library path
+            if not self.plumed_kernel_library:
+                # using "startswith" as recommended from https://docs.python.org/3/library/sys.html#sys.platform
+                if platform.startswith("linux"):
+                    self.plumed_kernel_library = os.path.join(
+                        os.environ["CONDA_PREFIX"], "lib", "libplumedKernel.so"
+                    )
+                elif platform.startswith("darwin"):
+                    self.plumed_kernel_library = os.path.join(
+                        os.environ["CONDA_PREFIX"], "lib", "libplumedKernel.dylib"
+                    )
+                else:
+                    logger.warning(
+                        f"Plumed is not supported with the '{platform}' platform."
+                    )
+
+            # Add Plumed kernel library to env dict
+            if not self.plumed_kernel_library:
+                if "PLUMED_KERNEL" not in os.environ.keys():
+                    os.environ = dict(
+                        os.environ, PLUMED_KERNEL=self.plumed_kernel_library
+                    )
+                else:
```

Comment:
Do you need this `else`? I think you can probably just clobber the environmental variable even if it's already defined. That is, in either case (whether it's there or not) just do:

```python
os.environ = dict(
                        os.environ, PLUMED_KERNEL=self.plumed_kernel_library
                    )
```

No? (I'm just guessing.)

Reply:
Just did a quick test and you're right I don't need to use `else`. I'll fix this.